oneclick.py

```diff
@@ -0,0 +1,600 @@
+            raise IOError("OneClick availability response not parseable - has no respdict.")
+
+        if not ('pageIndex' in respdict and 'pageCount' in respdict):
+            raise IOError("OneClick availability response not parseable - has no page counts.")
+        page_index = respdict['pageIndex']
+        page_count = respdict['pageCount']
+
+        while (page_count > (page_index+1)):
+            page_index += 1
+            response = self.search(availability='available', verbosity=self.RESPONSE_VERBOSITY[0], page_index=page_index)
+            tempdict = response.json()
+            if not ('items' in tempdict):
+                raise IOError("OneClick availability response not parseable - has no next dict.")
+            item_interest_pairs = tempdict['items']
+            respdict['items'].append(item_interest_pairs)
+
+        return respdict
+
+
+    def get_ebook_availability_info(self):
+        """
+        Gets a list of ebook items this library has access to, through the "availability" endpoint.
+        The response at this endpoint is laconic -- just enough fields per item to 
+        identify the item and declare it either available to lend or not.
+
+        :return A list of dictionary items, each item giving "yes/no" answer on a book's current availability to lend.
+        Example of returned item format:
+            "timeStamp": "2016-10-07T16:11:52.5887333Z"
+            "isbn": "9781420128567"
+            "mediaType": "eBook"
+            "availability": false
+            "titleId": 39764
+        """
+        url = "%s/libraries/%s/media/ebook/availability" % (self.base_url, str(self.library_id)) 
+
+        response = self.request(url)
+
+        resplist = response.json()
+        if not resplist:
+            raise IOError("OneClick availability response not parseable - has no resplist.")
+
+        return resplist
+
+
+    def get_metadata_by_isbn(self, identifier):
+        """
+        Gets metadata, s.a. publisher, date published, genres, etc for the 
+        ebook or eaudio item passed, using isbn to search on. 
+        If isbn is not found, the response we get from OneClick is an error message, 
+        and we throw an error.
+        """
+        if not identifier:
+            raise ValueError("Need valid identifier to get metadata.")
+
+        identifier_string = self.create_identifier_strings([identifier])[0]
+        url = "%s/libraries/%s/media/%s" % (self.base_url, str(self.library_id), identifier_string) 
+
+        response = self.request(url)
+
+        respdict = response.json()
+        if not respdict:
+            # should never happen
+            raise IOError("OneClick isbn search response not parseable - has no respdict.")
+
+        if "message" in respdict:
+            # can happen if searched for item that's not in library's catalog
+            raise ValueError("ISBN [%s] is not in library [%s]'s catalog." % (identifier_string, str(self.library_id)))
```

Comment:
What does the message say when you search  for an item not in the catalog? If there's a distinctive "not in the catalog" message, then check for that and either return None or raise a distinctive exception (which the calling code can check for). You can raise a more generic exception for other types of message.